src/shortestpaths/bellman-ford.jl

```diff
@@ -37,31 +37,35 @@ function bellman_ford_shortest_paths(
     distmx::AbstractMatrix{T}=weights(graph)
     ) where T<:Real where U<:Integer
 
-    active = Set{U}(sources)
-    sizehint!(active, nv(graph))
-    dists = fill(typemax(T), nv(graph))
-    parents = zeros(U, nv(graph))
+    nvg = nv(graph)
+    active = falses(nvg)
+    for s in sources
+        active[s] = true
+    end
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
     dists[sources] .= 0
     no_changes = false
+    new_active = falses(nvg)
 
-    for i in one(U):nv(graph)
+    for i in vertices(graph)
         no_changes = true
-        new_active = Set{U}()
-        for u in active
+        new_active .= false
+        for u in vertices(graph)[active]
             for v in outneighbors(graph, u)
                 relax_dist = distmx[u, v] + dists[u]
                 if dists[v] > relax_dist
                     dists[v] = relax_dist
                     parents[v] = u
                     no_changes = false
-                    push!(new_active, v)
+                    new_active[v] = true
                 end
             end
         end
         if no_changes
             break
         end
-        active = new_active
+        active .= new_active
```

Comment:
since you're resetting `new_active` to `false` in L53, why not just swap them? `active, new_active = new_active, active` - then you don't have to do any copying.

I *think* that should work :)